v5/v5_with_library/v5_stable.ino

```diff
@@ -759,33 +792,48 @@ void terminateLastMessage() {
   if (current_target_index == firstMessageIndex() || current_target_index <= 1)
     return;
   println("work index : " + String(current_target_index));
-  String temp_str = executeCommand(removeOk(readMessage(current_target_index)));
-  println("Last message [ " + temp_str + "]");
-  if (temp_str.indexOf("<executed>") != -1) {
+  String _message_ = (removeOk(readMessage(current_target_index)));
+  String mobileNumber =
+      getMobileNumberOfMsg(String(current_target_index), true);
+  if (senderIsAuthentic(mobileNumber, _message_))
+    _message_ = executeCommand(_message_);
+  else
+    println("Terminator has skiped message: {" + String(current_target_index) +
+            "} Because it is not supposed to be executed #771");
+  println("Last message [ " + _message_ + "]");
+  if (_message_.indexOf("<executed>") != -1) {
     deleteMessage(current_target_index);
     println("Message {" + String(current_target_index) + "} deleted");
-  } else { // if the message don't execute
-    String mobileNumber =
-        getMobileNumberOfMsg(String(current_target_index), false);
-    if (!checkStack(current_target_index)) {
-      if (!companyMsg(mobileNumber) &&
-          mobileNumber.indexOf("3374888420") == -1) {
-        // if its not company and self message
-        sendSMS("Unable to execute sms no. {" + String(current_target_index) +
-                "} message : [ " +
-                removeNewline(temp_str.substring(
-                    0, temp_str.indexOf(" <not executed>"))) +
+  } else {                                   // if the message don't execute
+    if (!checkStack(current_target_index)) { // ?
+      if (senderIsAuthentic(mobileNumber, _message_)) {
+        sendSMS("!Unable to execute previous sms no. {" +
+                String(current_target_index) + "} message : [ " +
+                removeNewline(_message_) + " ] from: {" + mobileNumber +
+                "} Removing it from stack so it don't disturb you !");
+        // TODO: add this message to orange-pi:
```

Comment:
you can add this as you have added logger if you have already completed you must remove this line other wise implement this functionality.

Reply:
i have removed this line as I'm already performing this action below.